File: binding/src/libqpagedpaintdevice.cpp
#include <QMarginsF>
#include <QPageLayout>
#include <QPageRanges>
#include <QPageSize>
#include <QPagedPaintDevice>
#include <QPaintDevice>
#include <qpagedpaintdevice.h>
#include "libqpagedpaintdevice.h"
#include "libqpagedpaintdevice.hxx"

bool QPagedPaintDevice_NewPage(QPagedPaintDevice* self) {
	return self->newPage();
}

bool QPagedPaintDevice_SetPageLayout(QPagedPaintDevice* self, const QPageLayout* pageLayout) {
	return self->setPageLayout(*pageLayout);
}

bool QPagedPaintDevice_SetPageSize(QPagedPaintDevice* self, const QPageSize* pageSize) {
	return self->setPageSize(*pageSize);
}

bool QPagedPaintDevice_SetPageOrientation(QPagedPaintDevice* self, int orientation) {
	return self->setPageOrientation(static_cast<QPageLayout::Orientation>(orientation));
}

bool QPagedPaintDevice_SetPageMargins(QPagedPaintDevice* self, const QMarginsF* margins, int units) {
	return self->setPageMargins(*margins, static_cast<QPageLayout::Unit>(units));
}

QPageLayout* QPagedPaintDevice_PageLayout(const QPagedPaintDevice* self) {
	return new QPageLayout(self->pageLayout());
}

void QPagedPaintDevice_SetPageRanges(QPagedPaintDevice* self, const QPageRanges* ranges) {
	self->setPageRanges(*ranges);
}

QPageRanges* QPagedPaintDevice_PageRanges(const QPagedPaintDevice* self) {
	return new QPageRanges(self->pageRanges());
}

// Base class handler implementation
bool QPagedPaintDevice_QBaseNewPage(QPagedPaintDevice* self) {
	auto* vqpagedpaintdevice = dynamic_cast<VirtualQPagedPaintDevice*>(self);
	if (vqpagedpaintdevice && vqpagedpaintdevice->isVirtualQPagedPaintDevice) {
vqpagedpaintdevice->setQPagedPaintDevice_NewPage_IsBase(true);
	return vqpagedpaintdevice->newPage();
}
}

// Auxiliary method to allow providing re-implementation
void QPagedPaintDevice_OnNewPage(QPagedPaintDevice* self, intptr_t slot) {
	auto* vqpagedpaintdevice = dynamic_cast<VirtualQPagedPaintDevice*>(self);
	if (vqpagedpaintdevice && vqpagedpaintdevice->isVirtualQPagedPaintDevice) {
vqpagedpaintdevice->setQPagedPaintDevice_NewPage_Callback(reinterpret_cast<VirtualQPagedPaintDevice::QPagedPaintDevice_NewPage_Callback>(slot));
}
}

// Base class handler implementation
bool QPagedPaintDevice_QBaseSetPageLayout(QPagedPaintDevice* self, const QPageLayout* pageLayout) {
	auto* vqpagedpaintdevice = dynamic_cast<VirtualQPagedPaintDevice*>(self);
	if (vqpagedpaintdevice && vqpagedpaintdevice->isVirtualQPagedPaintDevice) {
vqpagedpaintdevice->setQPagedPaintDevice_SetPageLayout_IsBase(true);
	return vqpagedpaintdevice->setPageLayout(*pageLayout);
}
}

// Auxiliary method to allow providing re-implementation
void QPagedPaintDevice_OnSetPageLayout(QPagedPaintDevice* self, intptr_t slot) {
	auto* vqpagedpaintdevice = dynamic_cast<VirtualQPagedPaintDevice*>(self);
	if (vqpagedpaintdevice && vqpagedpaintdevice->isVirtualQPagedPaintDevice) {
vqpagedpaintdevice->setQPagedPaintDevice_SetPageLayout_Callback(reinterpret_cast<VirtualQPagedPaintDevice::QPagedPaintDevice_SetPageLayout_Callback>(slot));
}
}

// Base class handler implementation
bool QPagedPaintDevice_QBaseSetPageSize(QPagedPaintDevice* self, const QPageSize* pageSize) {
	auto* vqpagedpaintdevice = dynamic_cast<VirtualQPagedPaintDevice*>(self);
	if (vqpagedpaintdevice && vqpagedpaintdevice->isVirtualQPagedPaintDevice) {
vqpagedpaintdevice->setQPagedPaintDevice_SetPageSize_IsBase(true);
	return vqpagedpaintdevice->setPageSize(*pageSize);
}
}

// Auxiliary method to allow providing re-implementation
void QPagedPaintDevice_OnSetPageSize(QPagedPaintDevice* self, intptr_t slot) {
	auto* vqpagedpaintdevice = dynamic_cast<VirtualQPagedPaintDevice*>(self);
	if (vqpagedpaintdevice && vqpagedpaintdevice->isVirtualQPagedPaintDevice) {
vqpagedpaintdevice->setQPagedPaintDevice_SetPageSize_Callback(reinterpret_cast<VirtualQPagedPaintDevice::QPagedPaintDevice_SetPageSize_Callback>(slot));
}
}

// Base class handler implementation
bool QPagedPaintDevice_QBaseSetPageOrientation(QPagedPaintDevice* self, int orientation) {
	auto* vqpagedpaintdevice = dynamic_cast<VirtualQPagedPaintDevice*>(self);
	if (vqpagedpaintdevice && vqpagedpaintdevice->isVirtualQPagedPaintDevice) {
vqpagedpaintdevice->setQPagedPaintDevice_SetPageOrientation_IsBase(true);
	return vqpagedpaintdevice->setPageOrientation(static_cast<QPageLayout::Orientation>(orientation));
}
}

// Auxiliary method to allow providing re-implementation
void QPagedPaintDevice_OnSetPageOrientation(QPagedPaintDevice* self, intptr_t slot) {
	auto* vqpagedpaintdevice = dynamic_cast<VirtualQPagedPaintDevice*>(self);
	if (vqpagedpaintdevice && vqpagedpaintdevice->isVirtualQPagedPaintDevice) {
vqpagedpaintdevice->setQPagedPaintDevice_SetPageOrientation_Callback(reinterpret_cast<VirtualQPagedPaintDevice::QPagedPaintDevice_SetPageOrientation_Callback>(slot));
}
}

// Base class handler implementation
bool QPagedPaintDevice_QBaseSetPageMargins(QPagedPaintDevice* self, const QMarginsF* margins, int units) {
	auto* vqpagedpaintdevice = dynamic_cast<VirtualQPagedPaintDevice*>(self);
	if (vqpagedpaintdevice && vqpagedpaintdevice->isVirtualQPagedPaintDevice) {
vqpagedpaintdevice->setQPagedPaintDevice_SetPageMargins_IsBase(true);
	return vqpagedpaintdevice->setPageMargins(*margins, static_cast<QPageLayout::Unit>(units));
}
}

// Auxiliary method to allow providing re-implementation
void QPagedPaintDevice_OnSetPageMargins(QPagedPaintDevice* self, intptr_t slot) {
	auto* vqpagedpaintdevice = dynamic_cast<VirtualQPagedPaintDevice*>(self);
	if (vqpagedpaintdevice && vqpagedpaintdevice->isVirtualQPagedPaintDevice) {
vqpagedpaintdevice->setQPagedPaintDevice_SetPageMargins_Callback(reinterpret_cast<VirtualQPagedPaintDevice::QPagedPaintDevice_SetPageMargins_Callback>(slot));
}
}

// Base class handler implementation
void QPagedPaintDevice_QBaseSetPageRanges(QPagedPaintDevice* self, const QPageRanges* ranges) {
	auto* vqpagedpaintdevice = dynamic_cast<VirtualQPagedPaintDevice*>(self);
	if (vqpagedpaintdevice && vqpagedpaintdevice->isVirtualQPagedPaintDevice) {
vqpagedpaintdevice->setQPagedPaintDevice_SetPageRanges_IsBase(true);
	vqpagedpaintdevice->setPageRanges(*ranges);
}
}

// Auxiliary method to allow providing re-implementation
void QPagedPaintDevice_OnSetPageRanges(QPagedPaintDevice* self, intptr_t slot) {
	auto* vqpagedpaintdevice = dynamic_cast<VirtualQPagedPaintDevice*>(self);
	if (vqpagedpaintdevice && vqpagedpaintdevice->isVirtualQPagedPaintDevice) {
vqpagedpaintdevice->setQPagedPaintDevice_SetPageRanges_Callback(reinterpret_cast<VirtualQPagedPaintDevice::QPagedPaintDevice_SetPageRanges_Callback>(slot));
}
}

// Derived class handler implementation
QPagedPaintDevicePrivate* QPagedPaintDevice_Dd(QPagedPaintDevice* self) {
	auto* vqpagedpaintdevice = dynamic_cast<VirtualQPagedPaintDevice*>(self);
	if (vqpagedpaintdevice && vqpagedpaintdevice->isVirtualQPagedPaintDevice) {
	return vqpagedpaintdevice->dd();
	} else {
	return self->QPagedPaintDevice::dd();
}
}

// Base class handler implementation
QPagedPaintDevicePrivate* QPagedPaintDevice_QBaseDd(QPagedPaintDevice* self) {
	auto* vqpagedpaintdevice = dynamic_cast<VirtualQPagedPaintDevice*>(self);
	if (vqpagedpaintdevice && vqpagedpaintdevice->isVirtualQPagedPaintDevice) {
vqpagedpaintdevice->setQPagedPaintDevice_Dd_IsBase(true);
	return vqpagedpaintdevice->dd();
}
}

// Auxiliary method to allow providing re-implementation
void QPagedPaintDevice_OnDd(QPagedPaintDevice* self, intptr_t slot) {
	auto* vqpagedpaintdevice = dynamic_cast<VirtualQPagedPaintDevice*>(self);
	if (vqpagedpaintdevice && vqpagedpaintdevice->isVirtualQPagedPaintDevice) {
vqpagedpaintdevice->setQPagedPaintDevice_Dd_Callback(reinterpret_cast<VirtualQPagedPaintDevice::QPagedPaintDevice_Dd_Callback>(slot));
}
}

void QPagedPaintDevice_Delete(QPagedPaintDevice* self) {
    delete self;
}

File: binding/include/libqpagedpaintdevice.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qpagedpaintdevice.html
pub const qpagedpaintdevice = struct {

    pub fn NewPage(self: ?*anyopaque, ) bool {
        return qtc.QPagedPaintDevice_NewPage(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnNewPage(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) bool) void {
        qtc.QPagedPaintDevice_OnNewPage(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseNewPage(self: ?*anyopaque, ) bool {
        return qtc.QPagedPaintDevice_QBaseNewPage(@ptrCast(self));
    }

    pub fn SetPageLayout(self: ?*anyopaque, pageLayout: ?*anyopaque) bool {
        return qtc.QPagedPaintDevice_SetPageLayout(@ptrCast(self), @ptrCast(pageLayout));
    }

    /// Allows for overriding the related default method
    pub fn OnSetPageLayout(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) bool) void {
        qtc.QPagedPaintDevice_OnSetPageLayout(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseSetPageLayout(self: ?*anyopaque, pageLayout: ?*anyopaque) bool {
        return qtc.QPagedPaintDevice_QBaseSetPageLayout(@ptrCast(self), @ptrCast(pageLayout));
    }

    pub fn SetPageSize(self: ?*anyopaque, pageSize: ?*anyopaque) bool {
        return qtc.QPagedPaintDevice_SetPageSize(@ptrCast(self), @ptrCast(pageSize));
    }

    /// Allows for overriding the related default method
    pub fn OnSetPageSize(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) bool) void {
        qtc.QPagedPaintDevice_OnSetPageSize(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseSetPageSize(self: ?*anyopaque, pageSize: ?*anyopaque) bool {
        return qtc.QPagedPaintDevice_QBaseSetPageSize(@ptrCast(self), @ptrCast(pageSize));
    }

    pub fn SetPageOrientation(self: ?*anyopaque, orientation: i32) bool {
        return qtc.QPagedPaintDevice_SetPageOrientation(@ptrCast(self), @intCast(orientation));
    }

    /// Allows for overriding the related default method
    pub fn OnSetPageOrientation(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) bool) void {
        qtc.QPagedPaintDevice_OnSetPageOrientation(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseSetPageOrientation(self: ?*anyopaque, orientation: i32) bool {
        return qtc.QPagedPaintDevice_QBaseSetPageOrientation(@ptrCast(self), @intCast(orientation));
    }

    pub fn SetPageMargins(self: ?*anyopaque, margins: ?*anyopaque, units: i32) bool {
        return qtc.QPagedPaintDevice_SetPageMargins(@ptrCast(self), @ptrCast(margins), @intCast(units));
    }

    /// Allows for overriding the related default method
    pub fn OnSetPageMargins(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, i32) callconv(.c) bool) void {
        qtc.QPagedPaintDevice_OnSetPageMargins(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseSetPageMargins(self: ?*anyopaque, margins: ?*anyopaque, units: i32) bool {
        return qtc.QPagedPaintDevice_QBaseSetPageMargins(@ptrCast(self), @ptrCast(margins), @intCast(units));
    }

    pub fn PageLayout(self: ?*anyopaque, ) QtC.QPageLayout {
        return qtc.QPagedPaintDevice_PageLayout(@ptrCast(self));
    }

    pub fn SetPageRanges(self: ?*anyopaque, ranges: ?*anyopaque) void {
        qtc.QPagedPaintDevice_SetPageRanges(@ptrCast(self), @ptrCast(ranges));
    }

    /// Allows for overriding the related default method
    pub fn OnSetPageRanges(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QPagedPaintDevice_OnSetPageRanges(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseSetPageRanges(self: ?*anyopaque, ranges: ?*anyopaque) void {
        qtc.QPagedPaintDevice_QBaseSetPageRanges(@ptrCast(self), @ptrCast(ranges));
    }

    pub fn PageRanges(self: ?*anyopaque, ) QtC.QPageRanges {
        return qtc.QPagedPaintDevice_PageRanges(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QPagedPaintDevice_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/pagedpaintdevice.html#types
pub const enums = struct {
    pub const PdfVersion = enum {
        pub const PdfVersion_1_4: i32 = 0;
        pub const PdfVersion_A1b: i32 = 1;
        pub const PdfVersion_1_6: i32 = 2;
        pub const PdfVersion_X4: i32 = 3;
    };

};
